client/src/App.js

```diff
@@ -8,32 +10,51 @@ import Authentication from './pages/Authentication';
 import BasicLayoutRoute from './layoutRoutes/BasicLayoutRoute';
 import HeaderBottomNavLayoutRoute from './layoutRoutes/HeaderBottomNavLayoutRoute';
 import HeaderSideBottomNavLayoutRoute from './layoutRoutes/HeaderSideBottomNavLayoutRoute';
+import useAuth from './contexts/auth/useAuth';
 
 function App() {
+  const { isAuthenticated } = useAuth();
+
   return (
-    <>
-      <Router>
-        <GlobalStyles />
-        <Switch>
-          <BasicLayoutRoute exact path="/login" component={Authentication} />
-          <BasicLayoutRoute exact path="/register" component={Authentication} />
-          <HeaderSideBottomNavLayoutRoute
-            exact
-            path="/popular"
-            component={Popular}
-            site="popular"
-          />
-          <HeaderBottomNavLayoutRoute exact path="/recs" component={Recs} />
-          <HeaderSideBottomNavLayoutRoute
-            exact
-            path="/lists"
-            component={Lists}
-            site="lists"
-          />
-          <BasicLayoutRoute path="/" Redirect to="/login" />
-        </Switch>
-      </Router>
-    </>
+    <UserProvider>
+      <AuthProvider>
+        <Router>
+          <GlobalStyles />
+          <Switch>
+            <BasicLayoutRoute exact path="/login" component={Authentication} />
+            <BasicLayoutRoute
+              exact
+              path="/register"
+              component={Authentication}
+            />
+            {isAuthenticated ? (
+              <>
+                <HeaderSideBottomNavLayoutRoute
+                  exact
+                  path="/popular"
+                  component={Popular}
+                  site="popular"
+                />
+                <HeaderBottomNavLayoutRoute
+                  exact
+                  path="/recs"
+                  component={Recs}
+                />
+                <HeaderSideBottomNavLayoutRoute
+                  exact
+                  path="/lists"
+                  component={Lists}
+                  site="lists"
+                />{' '}
+              </>
+            ) : (
+              <Redirect to="/" />
```

Comment:
Be careful when using `<Redirect>` directly in your `<Switch>`.
This might lead to an infinite redirection loop.
In your case, it might be possible that you are logged out and on `/` route. It tries to redirect to `/` again and again.

Better only use routes in your Switch.
https://reacttraining.com/react-router/web/api/Redirect